integration/healthcheck_test.go

```diff
@@ -42,45 +37,54 @@ func (s *HealthCheckSuite) TestSimpleConfiguration(c *check.C) {
 	defer cmd.Process.Kill()
 
 	// wait for traefik
-	err = utils.TryRequest("http://127.0.0.1:8080/api/providers", 60*time.Second, func(res *http.Response) error {
-		body, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			return err
-		}
-		if !strings.Contains(string(body), "Host:test.localhost") {
-			return errors.New("Incorrect traefik config: " + string(body))
-		}
-		return nil
-	})
+	err = try.GetRequest("http://127.0.0.1:8080/api/providers", 60*time.Second, try.BodyContains("Host:test.localhost"))
 	c.Assert(err, checker.IsNil)
 
 	client := &http.Client{}
-	req, err := http.NewRequest("GET", "http://127.0.0.1:8000/health", nil)
+	req, err := http.NewRequest(http.MethodGet, "http://127.0.0.1:8000/health", nil)
 	c.Assert(err, checker.IsNil)
 	req.Host = "test.localhost"
 
 	resp, err := client.Do(req)
 	c.Assert(err, checker.IsNil)
 	c.Assert(resp.StatusCode, checker.Equals, 200)
 
-	resp, err = client.Do(req)
+	// Fix all whoami health to 500
+	whoamiHosts := []string{whoami1Host, whoami2Host}
+	for _, whoami := range whoamiHosts {
+		healthReq, err := http.NewRequest(http.MethodPost, "http://"+whoami+"/health", bytes.NewBuffer([]byte("500")))
+		c.Assert(err, checker.IsNil)
+		_, err = client.Do(healthReq)
+		c.Assert(err, checker.IsNil)
+	}
+
+	// Waiting for Traefik healthcheck
+	try.Sleep(2 * time.Second)
+
+	// Verify frontend health : 500
+	err = try.Request(req, 3*time.Second, try.StatusCodeIs(500))
 	c.Assert(err, checker.IsNil)
-	c.Assert(resp.StatusCode, checker.Equals, 200)
 
-	healthReq, err := http.NewRequest("POST", "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("500")))
+	// Change one whoami health to 200
+	healthReq1, err := http.NewRequest(http.MethodPost, "http://"+whoami1Host+"/health", bytes.NewBuffer([]byte("200")))
 	c.Assert(err, checker.IsNil)
-	_, err = client.Do(healthReq)
+	_, err = client.Do(healthReq1)
 	c.Assert(err, checker.IsNil)
 
-	time.Sleep(time.Second * 3)
-
-	resp, err = client.Do(req)
+	// Verify frontend health : before
```

Comment:
I think the remaining block up until line 89 can be simplified as following:

1. Try-request for a 200 response code. Once we succeed, we know for sure that the first whoami host has successfully toggled its health status back to 200. Also, this way we don't have to rely on the `try.Sleep` in line 79.
1. Try-request for a 500 response code. This verifies that the second whoami host is still unhealthy.

Paraphrased, we can remove the section between line 74 and line 80.